src/Core/Services/Implementations/StripePaymentService.cs

```diff
@@ -739,58 +732,103 @@ public class StripePaymentService : IPaymentService
-                        storableSubscriber, subResponse?.LatestInvoiceId);
+                    if (chargeNow)
+                    {
+                        paymentIntentClientSecret = await PayInvoiceAfterSubscriptionChangeAsync(
+                            storableSubscriber, subResponse?.LatestInvoiceId);
+                    }
+                    else
+                    {
+                        var invoiceService = new InvoiceService();
+                        var invoice = await invoiceService.FinalizeInvoiceAsync(subResponse.LatestInvoiceId, new InvoiceFinalizeOptions
+                        {
+                            AutoAdvance = false,
+                        });
+                        await invoiceService.SendInvoiceAsync(invoice.Id, new InvoiceSendOptions());
+                        paymentIntentClientSecret = null;
+                    }
                 }
                 catch
                 {
                     // Need to revert the subscription
                     await subscriptionService.UpdateAsync(sub.Id, new SubscriptionUpdateOptions
                     {
-                        Items = new List<SubscriptionItemOptions>
-                        {
-                            new SubscriptionItemOptions
-                            {
-                                Id = storageItem?.Id,
-                                Plan = storagePlanId,
-                                Quantity = storageItem?.Quantity ?? 0,
-                                Deleted = (storageItem?.Id == null || (storageItem?.Quantity ?? 0) == 0)
-                                    ? true : (bool?)null
-                            }
-                        },
+                        Items = new List<SubscriptionItemOptions> { revertItemOptionsGenerator(sub) },
                         // This proration behavior prevents a false "credit" from
                         //  being applied forward to the next month's invoice
                         ProrationBehavior = "none",
                         CollectionMethod = collectionMethod,
+                        DaysUntilDue = daysUntilDue,
                     });
                     throw;
                 }
             }
 
-            // Change back the subscription collection method
-            if (collectionMethod != "send_invoice")
+            // Change back the subscription collection method and/or days until due
+            if (collectionMethod != "send_invoice" || daysUntilDue == null)
             {
                 await subscriptionService.UpdateAsync(sub.Id, new SubscriptionUpdateOptions
                 {
                     CollectionMethod = collectionMethod,
+                    DaysUntilDue = daysUntilDue,
                 });
             }
 
             return paymentIntentClientSecret;
+
+        }
+
+        private delegate SubscriptionItemOptions SubscriptionUpdateItemOptionsGenerator(Subscription sub);
```

Comment:
Feels like we should create another for reverting just to be explicit what we're doing.

Reply:
I didn't like the idea of having the exact same delegate for two different tasks, which is a good indicator your pattern is wrong. I switched over to more of a strategy pattern.